app/request.go

```diff
@@ -59,15 +86,122 @@ func NewRequestCommand(getRequestClientFn GetRequestClientFn) (CommandOut, error
+			},
+		}, {
+			Name:    "wait",
+			Usage:   "wait for the request complete",
+			Aliases: []string{"w"},
+			Flags: []cli.Flag{
+				&cli.StringFlag{
+					Name:     RequestIDFlagName,
+					Usage:    "the request-id of the request",
+					Aliases:  []string{"r"},
+					Required: true,
+				},
+			},
+			Action: func(ctx *cli.Context) error {
+				return c.waitOnRequest(ctx, "", ctx.String(RequestIDFlagName))
 			},
 		}},
 	}}, nil
 }
+
+func (c *RequestClient) waitOnRequest(ctx *cli.Context, operation string, requestID string) error {
+
+	ticker := time.NewTicker(time.Millisecond)
+	defer ticker.Stop()
+	timer := time.NewTimer(ctx.Duration(RequestTimeoutFlagName))
+	defer timer.Stop()
+
+	writer := uilive.New()
+	writer.Start()
+	defer writer.Stop()
+
+	var status *request.RequestStatus
+	defer func() {
+		if status != nil {
+			if err := PrintProto(status); err != nil {
+				fmt.Fprintf(writer, "failed to print status: %s", err)
+			}
+		}
+	}()
+loop:
+	for {
+		select {
+		case <-timer.C:
+			return fmt.Errorf("timed out waiting for request to complete, namespace=%s, requestID=%s, timeout=%s",
+				ctx.String(NamespaceFlagName),
+				requestID,
+				ctx.Duration(RequestTimeoutFlagName),
+			)
+		case <-ticker.C:
+			var err error
+			status, err = c.getRequestStatus(ctx, requestID)
+			if err != nil {
+				return err
+			}
+			switch status.State {
+			case request.STATE_FULFILLED:
+				break loop
+			case request.STATE_FAILED:
+				fmt.Fprintf(writer, "operation failed \n")
+				return fmt.Errorf("request failed: %s", status.FailureReason)
+			case request.STATE_CANCELLED:
+				fmt.Fprintf(writer, "operation cancelled\n")
+				return fmt.Errorf("request was cancelled: %s", status.FailureReason)
+			}
+			if operation != "" {
+				fmt.Fprintf(writer, "waiting for %s operation (requestId='%s') to finish, current state: %s\n",
+					operation, requestID, request.State_name[int32(status.State)])
+			} else {
+				fmt.Fprintf(writer, "waiting for request with id='%s' to finish, current state: %s\n",
+					requestID, request.State_name[int32(status.State)])
+			}
+			if status.CheckDuration == nil || status.CheckDuration.Seconds == 0 {
+				ticker.Reset(minCheckDurationTime) // min check duration is 1 second
+			} else {
+				ticker.Reset(time.Second * time.Duration(status.CheckDuration.Seconds))
+			}
+		}
+	}
+	if operation != "" {
+		fmt.Fprintf(writer, "%s operation completed successfully\n", operation)
+	} else {
+		fmt.Fprintf(writer, "request with id='%s' finished successfully\n", requestID)
+	}
+	return nil
+}
+
+func (c *RequestClient) HandleRequestStatus(
+	ctx *cli.Context,
+	operation string,
+	status *request.RequestStatus,
+) error {
+	if status == nil {
+		// status can be empty when the operation is cancelled
+		return nil
+	}
+	if ctx.Bool(WaitForRequestFlagName) {
+		return c.waitOnRequest(ctx, operation, status.RequestId)
+	}
+	if err := PrintProto(status); err != nil {
+		return err
+	}
+	fmt.Printf(
```

Comment:
I think we should allow these messages to be silenced for scripted use cases which expect JSON output on success. Perhaps `isatty` gets us this in the follow-up PR? Unsure if it covers all cases or not.

Reply:
I plan on addressing this comprehensively in the next PR. 
In the next PR i hope to add a tcld version checker, which notifies the user to update their `tcld` when it gets old.